### PR TITLE
Adding Variant Passage Flow Visualizations and Data

### LINKED ISSUE #33: Gene count summary across passages
Need to determine how variants within genes travel across passages. This will require the generation of a matrix that looks like this:

| Gene | Only F0 | Both | Only F5 |
| :------ | :---------: | :----: | :-------: |
| _KRAS_ | X | Y | Z |
| ... | | | |

I will also do something for the unique patients that have multiple mets and mouse mets sequenced

### PR DESCRIPTION
**Pull request by gwaygenomics**
Closes #33 

Two major notebooks are added in this PR

1. `mutations_across_passages.ipynb` - takes 8 groups of samples and outputs venn diagrams showing how variants are called across PDX passages. Also outputs tables describing the specific overlapping genes and cosmic IDs
2. `metastatic_passages.ipynb` - takes 2 groups of samples (both have unique relationships and several metastatic samples) and visualizes gene variant calls across several passages using [UpSetR](https://github.com/hms-dbmi/UpSetR)

The pull request also adds all the figures and tables that were generated by these two notebooks

### CODE REVIEW

File: scripts/nbconverted/mutations_across_passages.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+gene_depth_dict = {gene_id: [] for gene_id in set(unique_gene_df.index)}
+
+
+# ## Process Data and Output Venn Diagrams and Summary Tables
+
+# In[11]:
+
+
+total_counts = {}
+for sample_base in ['004', '005', '001', '006', '029', '030', '032', '040']:
```
Comment (danich1) [inline scripts/nbconverted/mutations_across_passages.py]: Not sure what these ids are for, but would it make sense to set this list as a variable for ease of readability?

File: scripts/nbconverted/mutations_across_passages.py [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+    cosmic_sub_df - where a specific variant is observed across samples
+    cosmic_passages - total number of passage types per sample base
+
+    Output:
+    The pattern of where the variant was observed
+    """
+
+    variant_sub_df = variant_sub_df.drop_duplicates(subset='sample_name')
+    num_var = len(set(variant_sub_df['sample_name']))
+    # Classify the passage type
```
Comment (huqiwen0313) [inline scripts/nbconverted/mutations_across_passages.py]: Add some description for the passage type ?

Comment (gwaygenomics) [reply scripts/nbconverted/mutations_across_passages.py]: Thanks @huqiwen0313 - I added to the description in the above docstring. Does it make sense?

**Comment by gwaygenomics** [comment]
Thanks @danich1 and @huqiwen0313 - ready again for rereview!

**Pull request merged**